### PR TITLE
Ret 3234

### PR DESCRIPTION
**Pull request by lvo700**
### JIRA link

https://tools.hmcts.net/jira/browse/RET-3234

### Change description

Apps & Case Mgmt common – view Respondents Applications to the tribunal from CitizenUI hub page

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/controllers/SupportingMaterialController.ts [hunk 1]
```diff
@@ -12,18 +11,12 @@ const logger = getLogger('SupportingMaterialController');
  */
 export default class SupportingMaterialController {
   public async get(req: AppRequest, res: Response): Promise<void> {
-    const userCase = req.session?.userCase;
-    let fileId = userCase?.contactApplicationFile?.document_url || userCase?.supportingMaterialFile?.document_url;
-    if (!fileId && userCase.selectedGenericTseApplication) {
-      fileId = userCase.selectedGenericTseApplication.value.documentUpload.document_url;
-    }
-
+    const docId = req.params.docId;
```
Comment (marcjudge) [inline src/main/controllers/SupportingMaterialController.ts:14]: I think we need to bring back some validation on this controller. I count three use cases so far. 

- Getting the supporting material added by the respondent to the selected application
- Getting the supporting material added by the claimant in the response to the selected application
- Getting the sendNotificationUploadDocument from the selectedRequestOrOrder

Similar to the validation on Line 21 of the deleted TribunalOrderDocumentController, use the docId provided to check if the document belongs to the case. If any of the three conditions are true, continue to make the call to the CaseApi... else return not found.

**Review by marcjudge — CHANGES_REQUESTED** [review]

File: src/main/controllers/helpers/DocumentHelpers.ts [hunk 2]
```diff
@@ -37,32 +37,20 @@ export const getDocumentAdditionalInformation = async (doc: Document, accessToke
...
-        "<a href='/getSupportingMaterial' target='_blank' class='govuk-link'>" +
-        file.document_filename +
-        '(' +
-        mimeType +
-        ', ' +
-        formatBytes(file.document_size) +
-        ')' +
-        '</a>';
-    }
+    downloadLink =
```
Comment (paul-peasegood) [inline src/main/controllers/helpers/DocumentHelpers.ts:44]: Could this be rewritten using template literals?

File: src/main/controllers/RespondentApplicationDetailsController.ts [hunk 3]
```diff
@@ -52,9 +52,66 @@ export default class RespondentApplicationDetailsController {
       }
     }
 
+    let claimantResponseDocDownload = undefined;
+
+    if (selectedApplication.value?.respondCollection && selectedApplication.value?.respondCollection.length) {
```
Comment (JackReevies) [inline src/main/controllers/RespondentApplicationDetailsController.ts:57]:
```suggestion
    if (selectedApplication.value?.respondCollection?.length) {
```

File: src/main/controllers/RespondentApplicationDetailsController.ts [hunk 4]
```diff
@@ -52,9 +52,66 @@ export default class RespondentApplicationDetailsController {
...
+      for (let i = selectedApplication.value?.respondCollection.length - 1; i >= 0; i--) {
+        if (
+          selectedApplication.value?.respondCollection[i].value.from === 'Claimant' &&
```
Comment (paul-peasegood) [inline src/main/controllers/RespondentApplicationDetailsController.ts:60]: Could we put selectedApplication.value.respondCollection[i].value in a variable, as we are reusing it?

Do we need the null check here as it is already done above?

File: src/main/controllers/RespondentApplicationDetailsController.ts [hunk 5]
```diff
@@ -52,9 +52,66 @@ export default class RespondentApplicationDetailsController {
...
+        }
+      }
+    }
+
+    if (claimantResponseDocDownload) {
+      try {
+        await getDocumentAdditionalInformation(claimantResponseDocDownload, req.session.user?.accessToken);
+      } catch (err) {
+        logger.error(err.message);
+        return res.redirect('/not-found');
```
Comment (JackReevies) [inline src/main/controllers/RespondentApplicationDetailsController.ts:74]: Why do we redirect to "not-found" if an error occurred during processing?

File: src/main/controllers/RespondentApplicationDetailsController.ts [hunk 6]
```diff
@@ -52,9 +52,66 @@ export default class RespondentApplicationDetailsController {
...
+            await getDocumentAdditionalInformation(decisionDocDownload[i], req.session.user?.accessToken);
+          } catch (err) {
+            logger.error(err.message);
+            return res.redirect('/not-found');
+          }
+          decisionDocDownloadLink[i] = createDownloadLink(decisionDocDownload[i]);
+        }
+      }
+      decisionContent = getTseApplicationDecisionDetails(selectedApplication, translations, decisionDocDownloadLink);
+    }
```
Comment (JackReevies) [inline src/main/controllers/RespondentApplicationDetailsController.ts:105]: Could these segments be split out into their own method? This method is over 100 lines long, seems like its complexity would be high

File: src/test/unit/views/respondent-application-details.test.ts [hunk 7]
```diff
@@ -14,17 +14,39 @@ const respondentApplicationDetailsJSONRaw = fs.readFileSync(
 
 const respondentApplicationDetailsJSON = JSON.parse(respondentApplicationDetailsJSONRaw);
 const titleClass = 'govuk-heading-xl govuk-!-margin-bottom-4 govuk-!-margin-top-1';
-const summaryListClass = 'govuk-summary-list__value';
-const summaryListHeadingClass = 'govuk-summary-list__key';
+const summaryListValueClass = 'govuk-summary-list__value';
+const summaryListKeyClass = 'govuk-summary-list__key';
+const summaryListTitleClass = 'govuk-summary-list__key govuk-heading-m govuk-!-margin-top-1';
+//const buttonClass = 'govuk-button';
```
Comment (JackReevies) [inline src/test/unit/views/respondent-application-details.test.ts:20]: commented code should be removed if not in use?

File: src/main/controllers/SupportingMaterialController.ts [hunk 8]
```diff
@@ -12,18 +11,12 @@ const logger = getLogger('SupportingMaterialController');
  */
 export default class SupportingMaterialController {
```
[the remark this answers is not in the document]
Comment (lvo700) [reply src/main/controllers/SupportingMaterialController.ts]: Renamed to AttachmentController

File: src/main/controllers/SupportingMaterialController.ts (hunk 1 above)
Comment (lvo700) [reply src/main/controllers/SupportingMaterialController.ts]: Added validation

File: src/main/controllers/helpers/DocumentHelpers.ts (hunk 2 above)
Comment (lvo700) [reply src/main/controllers/helpers/DocumentHelpers.ts:44]: Added template literal

File: src/main/controllers/RespondentApplicationDetailsController.ts (hunk 4 above)
Comment (lvo700) [reply src/main/controllers/RespondentApplicationDetailsController.ts]: Refactored

File: src/main/controllers/RespondentApplicationDetailsController.ts (hunk 3 above)
Comment (lvo700) [reply src/main/controllers/RespondentApplicationDetailsController.ts]: Refactored

File: src/main/controllers/RespondentApplicationDetailsController.ts (hunk 5 above)
Comment (lvo700) [reply src/main/controllers/RespondentApplicationDetailsController.ts:68]: Spoke to Mehmet. This is what we have for now. Not being able to find documents is a bug and he's currently working on it.

File: src/main/controllers/RespondentApplicationDetailsController.ts (hunk 6 above)
Comment (lvo700) [reply src/main/controllers/RespondentApplicationDetailsController.ts:100]: Moved claimant response doc download to helper

File: src/test/unit/views/respondent-application-details.test.ts (hunk 7 above)
Comment (lvo700) [reply src/test/unit/views/respondent-application-details.test.ts]: Removed comment

**Review by ikirsanov — APPROVED** [review]

File: src/main/controllers/AttachmentController.ts [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+  public async get(req: AppRequest, res: Response): Promise<void> {
+    const docId = req.params.docId;
+    const userCase = req.session?.userCase;
+    let respondentAppDocId;
+    let claimantResponseDocId;
+    let decisionDocId;
+    let requestDocId;
+    let contactTribunalDocId;
+
+    if (req.session.documentDownloadPage === PageUrls.RESPONDENT_APPLICATION_DETAILS) {
```
Comment (marcjudge) [inline src/main/controllers/AttachmentController.ts:26]: Not sure if this is the best way of doing things, but I don't want to block this any longer

**Review by marcjudge — APPROVED** [review]

**Pull request merged**